File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    
    hash = {}
    for i in participant:
        if i in hash:
            hash[i] += 1
        else:
            hash[i] = 1
            
            
    for i in completion:
        if hash[i] == 1:
            del hash[i]
        else:
            hash[i] -= 1
            
    for i, count in hash.items():
        if count > 0:
            return i
        
    return answer

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <bits/stdc++.h>
using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    
    if(participant.size() == 1) {
        if(completion.size() == 0) return participant[0];
    }
    sort(participant.begin(),participant.end());
    sort(completion.begin(),completion.end());
    
    for(int i=0; i<completion.size(); i++) {
        if(completion[i] != participant[i]) {
            answer = participant[i];
            return answer;
        }  
    }
}